app/components/preprint-status-banner/component.js

```diff
@@ -61,13 +61,18 @@ export default Ember.Component.extend({
         if (this.get('submission.provider.reviewsCommentsPrivate')) {
             this.set('latestAction', null);
         } else {
-            this.get('submission.actions').then(actions => {
-                if (actions.length) {
-                    this.set('latestAction', actions.get('firstObject'));
-                } else {
-                    this.set('latestAction', null);
-                }
-            });
+            const submissionActions = this.get('submission.actions');
+            if (submissionActions) {
```

Comment:
Is `submission.actions` null sometimes? I'm curious when that happens.

Reply:
the main reason is to make the code easier to test, previously unit tests were failing with submission.actions is not defined.  Also assuming that actions are correctly retrieved might possibly introduce error in the future.

Reply:
`submission.actions` should not be undefined... `hasMany` relations on models should return a promise-like object. Did you include `model:action` in the `needs` for that test?

If there's an error retrieving, the returned promise-like object will reject/fail. The way to handle that would be a `.catch()`.

Reply:
What about when `submission` is undefined? That's what's happening in the unit test and theoretically could happen in real-world use of this component depending on where the `submission` parameter comes from. The component shouldn't make an assumption that `submission` is always defined (or that it's necessarily a preprint if we want to make it really robust).

Reply:
Hmm, fair point. In practice it doesn't make sense to even render this component without an actual `submission`, but you're right it should be able to handle that case.